x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/planner/QueryTranslatorTests.java

```diff
@@ -2508,4 +2508,92 @@ public void testAddMissingEqualsToNestedBoolField() {
 
         assertEquals(expectedCondition, condition);
     }
+
+    public void testSubqueryBasicSelect() throws Exception {
+        PhysicalPlan p = optimizeAndPlan("SELECT int FROM " +
+            "( SELECT int FROM test )");
+    }
+
+    public void testSubquerySelectOnFieldAlias() throws Exception {
+        PhysicalPlan p = optimizeAndPlan("SELECT i FROM " +
+            "( SELECT int AS i FROM test )");
+    }
+
+    public void testSubqueryGroupByNoAlias() throws Exception {
+        PhysicalPlan p = optimizeAndPlan("SELECT int FROM " +
+            "( SELECT int FROM test ) " +
+            "GROUP BY int");
+    }
+
+    public void testSubqueryGroupByOnFieldAlias() throws Exception {
+        PhysicalPlan p = optimizeAndPlan("SELECT i FROM " +
+            "( SELECT int AS i FROM test ) " +
+            "GROUP BY i");
+    }
+
+    public void testSubqueryFilterOrderByAlias() throws Exception {
+        PhysicalPlan p = optimizeAndPlan("SELECT i FROM " +
+            "( SELECT int AS i FROM test ) " +
+            "WHERE i IS NOT NULL " +
+            "ORDER BY i");
+    }
+
+    @AwaitsFix(bugUrl = "follow-up to https://github.com/elastic/elasticsearch/pull/67216")
+    public void testSubqueryGroupByFilterAndOrderByByAlias() throws Exception {
+        PhysicalPlan p = optimizeAndPlan("SELECT i FROM " +
+            "( SELECT int AS i FROM test ) " +
+            "WHERE i IS NOT NULL " +
+            "GROUP BY i " +
+            "ORDER BY i");
+    }
```

Comment:
Same issue here in resolving the reference in the aggregate hidden by the filter.